client/verta/verta/deployment/_deployedmodel.py

```diff
@@ -255,6 +255,19 @@ def predict(self, x, compress=False, max_retries=5, always_retry_404=True, alway
         _utils.raise_for_http_error(response)
 
 
+def prediction_input_df(func):
+    def prediction(self, input):
+        import pandas as pd
+
+        if isinstance(input, pd.DataFrame):
+            # input is dataframe.
+            return func(self, input)
+        else:
+            # input is list of records
+            return func(self, pd.DataFrame.from_records(input))
```

Comment:
This is to allow user to pass in data frame to model.predict as usual (instead of having to convert it into records). The conversion into list of records (and back) will be done behind the scene by Verta.